src/Custom/ChatModelFactory.cs

```diff
@@ -0,0 +1,253 @@
+            finishReason: finishReason,
+            content: content,
+            refusal: refusal,
+            toolCalls: toolCalls,
+            role: role,
+            functionCall: default,
+            contentTokenLogProbabilities: contentTokenLogProbabilities,
+            refusalTokenLogProbabilities: refusalTokenLogProbabilities,
+            createdAt: createdAt,
+            model: model,
+            systemFingerprint: systemFingerprint,
+            usage: usage);
+    }
+
+    /// <summary> Initializes a new instance of <see cref="OpenAI.Chat.StreamingChatCompletionUpdate"/>. </summary>
+    /// <returns> A new <see cref="OpenAI.Chat.StreamingChatCompletionUpdate"/> instance for mocking. </returns>
+    public static StreamingChatCompletionUpdate StreamingChatCompletionUpdate(
+        string completionId = null,
+        ChatMessageContent contentUpdate = null,
+        IEnumerable<StreamingChatToolCallUpdate> toolCallUpdates = null,
+        ChatMessageRole? role = default,
+        string refusalUpdate = null,
+        IEnumerable<ChatTokenLogProbabilityDetails> contentTokenLogProbabilities = null,
+        IEnumerable<ChatTokenLogProbabilityDetails> refusalTokenLogProbabilities = null,
+        ChatFinishReason? finishReason = default,
+        DateTimeOffset createdAt = default,
+        string model = null,
+        string systemFingerprint = null,
+        ChatTokenUsage usage = default)
+    {
+        return OpenAI.Chat.OpenAIChatModelFactory.StreamingChatCompletionUpdate(
+            completionId: completionId,
+            contentUpdate: contentUpdate,
+            functionCallUpdate: default,
+            toolCallUpdates: toolCallUpdates,
+            role: role,
+            refusalUpdate: refusalUpdate,
+            contentTokenLogProbabilities: contentTokenLogProbabilities,
+            refusalTokenLogProbabilities: refusalTokenLogProbabilities,
+            finishReason: finishReason,
+            createdAt: createdAt,
+            model: model,
+            systemFingerprint: systemFingerprint,
+            usage: usage);
+    }
+
+    /// <summary> Initializes a new instance of <see cref="OpenAI.Chat.ChatCompletionDeletionResult"/>. </summary>
+    /// <returns> A new <see cref="OpenAI.Chat.ChatCompletionDeletionResult"/> instance for mocking. </returns>
+    public static ChatCompletionDeletionResult ChatCompletionDeletionResult(
+        bool deleted = true, 
+        string chatCompletionId = null)
+    {
+        return new ChatCompletionDeletionResult(
+            deleted: deleted,
+            @object: "chat.completion.deleted",
+            chatCompletionId: chatCompletionId,
+            additionalBinaryDataProperties: null);
+    }
+
+    /// <summary> Initializes a new instance of <see cref="OpenAI.Chat.UserChatMessage"/>. </summary>
+    /// <returns> A new <see cref="OpenAI.Chat.UserChatMessage"/> instance for mocking. </returns>
+    public static UserChatMessage UserChatMessage(
```

Comment:
User chat message can be instantiated using public constructor and it seems like all its properties can be initialized with public APIs too. Therefore, there does not seem to be a need for the model factory method. Can you review all the types that you added factory method for and make sure we are not adding factory methods that are not necessary